src/Components/Web/src/Forms/InputSelect.cs

```diff
@@ -24,16 +27,51 @@ public class InputSelect<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTy
         /// </summary>
         [DisallowNull] public ElementReference? Element { get; protected set; }
 
+        /// <summary>
+        /// Gets or sets the current value of the input as a string array.
+        /// </summary>
+        protected string?[]? CurrentValueAsStringArray
+        {
+            get
+            {
+                if (CurrentValue is not Array array)
+                {
+                    return null;
+                }
+                
+                return array
+                    .Cast<object?>()
+                    .Select(item => item?.ToString())
+                    .ToArray();
+            }
+            set
+            {
+                CurrentValue = BindConverter.TryConvertTo<TValue>(value, CultureInfo.InvariantCulture, out var result)
+                    ? result
+                    : default;
+            }
+        }
+
         /// <inheritdoc />
         protected override void BuildRenderTree(RenderTreeBuilder builder)
         {
             builder.OpenElement(0, "select");
             builder.AddMultipleAttributes(1, AdditionalAttributes);
             builder.AddAttribute(2, "class", CssClass);
-            builder.AddAttribute(3, "value", BindConverter.FormatValue(CurrentValueAsString));
-            builder.AddAttribute(4, "onchange", EventCallback.Factory.CreateBinder<string?>(this, __value => CurrentValueAsString = __value, CurrentValueAsString));
-            builder.AddElementReferenceCapture(5, __selectReference => Element = __selectReference);
-            builder.AddContent(6, ChildContent);
+
+            if (AdditionalAttributes?.ContainsKey("multiple") ?? false)
+            {
+                builder.AddAttribute(3, "value", BindConverter.FormatValue(CurrentValue)?.ToString());
+                builder.AddAttribute(4, "onchange", EventCallback.Factory.CreateBinder(this, __value => CurrentValueAsStringArray = __value, CurrentValueAsStringArray));
+            }
+            else
+            {
+                builder.AddAttribute(5, "value", CurrentValueAsString);
+                builder.AddAttribute(6, "onchange", EventCallback.Factory.CreateBinder<string?>(this, __value => CurrentValueAsString = __value, CurrentValueAsString));
+            }
+
+            builder.AddElementReferenceCapture(7, __selectReference => Element = __selectReference);
+            builder.AddContent(8, ChildContent);
```

Comment:
This might seem a bit weird, but this was the best way I could think to interact with the edit context when binding array values. If we bind to `CurrentValue` directly, parsing failures in the `BindConverter` get lost.

Reply:
If someone else has a better suggestion for this, I'm all ears.

Reply:
If you're referring to having two different `EventCallback.Factory.CreateBinder` calls, this seems totally fine to me.

Slightly broader thought though: do we need the developer to specify `multiple`? Would we be able to infer that based on whether they are binding to an array?

Reply:
Interesting idea - I wonder if we could ignore any user-specified `multiple` attribute entirely and as you suggested infer `multiple`. We might not even really need to explicitly handle the case in which the developer wrongly includes/excludes the `multiple` attribute.